contracts/Marketplace.sol

```diff
@@ -1,74 +1,272 @@
-pragma solidity ^0.4.24;
-
-import "openzeppelin-solidity/contracts/utils/Address.sol";
-import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
-import "openzeppelin-solidity/contracts/lifecycle/Pausable.sol";
-
-contract Service is Ownable, Pausable {
-    bytes32 public sid;
-    bytes32 public latest;
-    mapping (bytes32 => Version) public versions;
-    mapping (bytes32 => mapping(address => bool)) private _versionAccess;
-
-    struct Version {
-        bytes32 id;
-        bytes32 location;
-        uint256 price;
-        bool _exists;
-    }
+pragma solidity >=0.5.0 <0.6.0;
 
-    event VersionCreated(bytes32 sid, bytes32 id, bytes32 location, uint256 price);
+import "../node_modules/openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "../node_modules/openzeppelin-solidity/contracts/lifecycle/Pausable.sol";
 
-    constructor(address creator, bytes32 _sid) Ownable() Pausable() public {
-        sid = _sid;
-        transferOwnership(creator);
-        addPauser(creator);
-        renouncePauser();
-    }
+contract Marketplace is Ownable, Pausable {
+  // ------------------------------------------------------
+  // Structures
+  // ------------------------------------------------------
 
-    function createVersion(bytes32 versionHash, bytes32 location, uint256 price) public onlyOwner whenNotPaused {
-        require(!versions[versionHash]._exists, "This version already exists");
-        versions[versionHash] = Version({
-            id: versionHash,
-            location: location,
-            price: price,
-            _exists: true
-        });
-        latest = versionHash;
-        emit VersionCreated(sid, versionHash, location, price);
-        return;
-    }
+  struct Service {
+    address owner;
+    string sid;
+    Version[] versions;
+    Offer[] offers;
+    Payment[] payments;
+  }
+
+  struct Version {
+    string hash;
+    string url;
+  }
+
+  struct Offer {
+    uint price;
+    // address payment; // TODO: to implement later
+    address payable seller;
+    bool active;
```

Comment:
Remove active and just change the address of seller to 0 - storage save && less cost, or maybe change price to 0.

Reply:
The goal is to keep an historic of the offers.
I will simplify and include only 1 offer!

Reply:
Ok, so i need to read about memory layout in struct. maybe address should be first filed